core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -323,6 +326,7 @@ private[spark] class CoarseGrainedExecutorBackend(
               // move forward.
               lastTaskRunningTime = System.nanoTime()
             }
+            Thread.sleep(sleep_time)
```

Comment:
This was moved so initial sleep time didn't have sleep_time added to it on the first pass through right? Nothing else?

Reply:
Yeah. No semantic change. We are still by default waiting for sleep_time the first time and the last time around (it is an infinite while loop that can only exit via an `exit(1)` -- via process death). I just wanted the first sleep interval to be configurable for testing. But no production change to the shutdown thread.